argilla-server/src/argilla_server/api/schemas/v1/users.py

```diff
@@ -35,16 +36,15 @@ class User(BaseModel):
     inserted_at: datetime
     updated_at: datetime
 
-    class Config:
-        orm_mode = True
+    model_config = ConfigDict(from_attributes=True)
 
 
 class UserCreate(BaseModel):
-    first_name: constr(min_length=1, strip_whitespace=True)
-    last_name: Optional[constr(min_length=1, strip_whitespace=True)]
     username: str = Field(..., min_length=1)
-    role: Optional[UserRole]
     password: str = Field(min_length=USER_PASSWORD_MIN_LENGTH, max_length=USER_PASSWORD_MAX_LENGTH)
+    first_name: constr(min_length=1, strip_whitespace=True)
+    last_name: Optional[constr(min_length=1, strip_whitespace=True)] = None
```

Comment:
Maybe we should start using annotated `str` with [`StringConstraints`](https://docs.pydantic.dev/latest/api/types/#pydantic.types.StringConstraints) instead of using the old `constr`.

This applies to all the `constr` used in the PR.

So something like:

```python
last_name: Optional[Annotated[str, StringConstraints(min_length=1, strip_whitespace=True)]] = None
```

You can take a look to https://github.com/argilla-io/argilla/pull/5508/files#diff-0e72eee9f8cea03b100e007fe81d5a97c5ebea93955187f083bb908a4cee2e29R48